projects/packages/forms/src/store/integrations/reducer.ts

```diff
@@ -0,0 +1,55 @@
+import {
+	RECEIVE_INTEGRATIONS,
+	INVALIDATE_INTEGRATIONS,
+	SET_INTEGRATIONS_LOADING,
+	SET_INTEGRATIONS_ERROR,
+} from './action-types';
+import type { IntegrationsState, IntegrationsAction } from './types';
+
+const DEFAULT_STATE: IntegrationsState = {
+	items: null,
+	isLoading: false,
+	error: null,
+};
+
+/**
+ * Integrations store reducer.
+ *
+ * @param state  - Current state
+ * @param action - Dispatched action
+ * @return Updated state
+ */
+export default function reducer(
+	state: IntegrationsState = DEFAULT_STATE,
+	action: IntegrationsAction
+): IntegrationsState {
+	switch ( action.type ) {
+		case SET_INTEGRATIONS_LOADING:
+			return {
+				...state,
+				isLoading: !! action.isLoading,
```

Comment:
This is a bit of a semantic issue. I am not really sure what the correct way is so feel free to ignore it. 
I just noticed that in gutenberg we sometime use isResolving see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#isresolving 

Should we use that instead of isLoading?

Reply:
Good thought, but I think this wouldn't quite work. My understanding is that isResolving only tracks when resolvers run for a specific selector (e.g., getIntegrations). It wouldn't cover the refreshIntegrations 'thunk' that we added in this PR so we can actively trigger refreshes, so we'd lose loading state during those refreshes I think. And the refreshes are pretty key in a lot of places.